agent_scheduler/task_helpers.py

```diff
@@ -142,32 +160,51 @@ def deserialize_img2img_image_args(args: dict):
 
 def serialize_controlnet_args(cnet_unit):
     args: dict = cnet_unit.__dict__
-    args["is_cnet"] = True
+    new_args = {}
+    new_args["is_cnet"] = True
     for k, v in args.items():
-        if k == "image" and v is not None:
-            args[k] = {
-                "image": serialize_image(v["image"]),
-                "mask": serialize_image(v["mask"])
-                if v.get("mask", None) is not None
-                else None,
-            }
-        if isinstance(v, Enum):
-            args[k] = v.value
+        if k == 'image':
+            if hasattr(v, "image") and v.image is not None:
+                new_args[k] = {
+                    "image": serialize_image(v["image"]),
+                    "mask": serialize_image(v["mask"])
+                    if v.get("mask", None) is not None
+                    else None,
+                }
+            elif type(v) is dict and v.get('image', None) is not None:
+                new_args[k] = {
+                    "image": serialize_image(v),
+                    "mask": None if v.get("mask", None) is None else serialize_image(v["mask"]),
+                }
+            else:
+                new_args[k] = serialize_image(v)
+        elif k == 'image' and isinstance(v, (np.ndarray, torch.Tensor, Image.Image)):
```

Comment:
The `elif` will never be executed. L180 already covers this case so it's safe to remove this.